### ISSUE TITLE
Solid milestones stop updating on an out-of-sync node once getTransactionsToApprove is received

### ISSUE THREAD

**Issue by rajivshah3** [iotaledger/iri#844]
### Bug description
My node is out-of-sync with the network. If I turn off `nginx`, therefore preventing API requests from coming to my node, the solid milestones will gradually increase as usual. If I turn on `nginx` and receive a `getTransactionsToApprove` request, the solid milestones will stop updating and resource usage increases significantly.

[Logs during getTransactionsToApprove request](https://pastebin.com/GrmtPCZR)

Before `getTransactionsToApprove`,  IRI used 70% CPU and 6.88 GB RAM. After `getTransactionsToApprove`, IRI used 262% CPU and 9.34 GB RAM.

### Hardware Spec
Ubuntu 16.04, 4vCPUs and 12GB RAM

### Steps To Reproduce
1. Set up IRI in a way that will allow you to block and unblock API requests without restarting IRI (using `nginx` proxy pass, for example)
2. Block API requests and then start IRI
3. Allow it to run for a few minutes. The solid milestones should be increasing
4. Unblock API requests and send a `getTransactionsToApprove` request
5. The solid milestones will stop updating, even after 5 minutes

### Expected behavior
Normal behavior, solid milestones should gradually increase

### Actual behavior
Solid milestones stop increasing after `getTransactionsToApprove`. The [transaction](https://thetangle.org/transaction/SDREQJXFDDFPMMORVAOOEIHXZOFBHINMJUJJLQMQLCOETTEXQTTTFFLYVJKCXHMREZDJSXEJBVJPZ9999) that IRI began looking for tips was a milestone issued almost 2 days ago. I'm not sure if this is the expected behavior.

### Errors
Upon stopping IRI:
```
06/28 22:07:56.479 [Shutdown Hook] DEBUG io.undertow - stopping undertow server io.undertow.Undertow@7d8995e
06/28 22:07:57.239 [pool-3-thread-4] INFO  com.iota.iri.network.Node - Shutting down Process Received Data Thread
06/28 22:07:57.240 [pool-3-thread-5] INFO  com.iota.iri.network.Node - Shutting down Reply To Request Thread
06/28 22:07:57.240 [pool-3-thread-1] INFO  com.iota.iri.network.Node - Shutting down Broadcaster Thread
06/28 22:07:57.479 [pool-3-thread-3] INFO  com.iota.iri.network.Node - Shutting down Neighbor DNS Refresher Thread
06/28 22:08:00.472 [pool-3-thread-2] INFO  com.iota.iri.network.Node - Shutting down Requester Thread
06/28 22:08:00.475 [UDP receiving thread] INFO  com.iota.iri.network.UDPReceiver - Shutting down spawning Receiver Thread
Exception in thread "Thread-2" java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@23db4556 rejected from java.util.concurrent.ThreadPoolExecutor@61655764[Shutting down, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 292]
        at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
        at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
        at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
        at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
        at com.iota.iri.network.replicator.ReplicatorSourcePool.run(ReplicatorSourcePool.java:50)
        at java.lang.Thread.run(Thread.java:748)
06/28 22:09:00.061 [Shutdown Hook] INFO  com.iota.iri.storage.Tangle - Shutting down Tangle Persistence Providers... 
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000000000000030, pid=3158, tid=0x00007f9b2d3d3700
#
# JRE version: Java(TM) SE Runtime Environment (8.0_171-b11) (build 1.8.0_171-b11)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.171-b11 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  0x0000000000000030
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# hs_err_pid3158.log
Compiled method (nm) 4453099 1669     n 0       org.rocksdb.RocksDB::get (native)
 total in heap  [0x00007f9bc551ebd0,0x00007f9bc551ef68] = 920
 relocation     [0x00007f9bc551ecf8,0x00007f9bc551ed40] = 72
 main code      [0x00007f9bc551ed40,0x00007f9bc551ef60] = 544
 oops           [0x00007f9bc551ef60,0x00007f9bc551ef68] = 8
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
Aborted (core dumped)
```

**Comment by alon-e** [comment]
A solution could be to redefine `invalidSubtangleStatus()` to check that a node is "reasonably solid" to perform the required task.
Currently, the definition is "not solid on any milestone" - i.e. `latestSolidSubtangleMilestone`: `999...`
This could be changed to `latestSolidSubtangleMilestone + max_depth < latestMilestone`  (the predicate is a negative).

I'll submit a PR and start testing.

**Comment by jakubcech** [comment]
This should be now fixed by #1533. Will be in the next release.
